lintcheck/src/main.rs

```diff
@@ -234,9 +252,15 @@ fn normalize_diag(
 }
 
 /// Builds clippy inside the repo to make sure we have a clippy executable we can use.
-fn build_clippy() -> String {
+fn build_clippy(release_build: bool) -> String {
     let output = Command::new("cargo")
-        .args(["run", "--bin=clippy-driver", "--", "--version"])
+        .args([
+            "run",
+            "--bin=clippy-driver",
+            if release_build { "-r" } else { "" },
+            "--",
+            "--version",
+        ])
         .stderr(Stdio::inherit())
```

Comment:
Would we want to bump up `CARGO_PROFILE_RELEASE_DEBUG` when perf is enabled too?